packages/api/src/payments/payrexxSubscriptionPaymentProvider.ts

```diff
@@ -0,0 +1,449 @@
+  externalId: string
+) {
+  return subscriptionClient.findFirst({
+    where: {
+      properties: {
+        some: {
+          key: 'payrexx_external_id',
+          value: `${externalId}`
+        }
+      }
+    },
+    include: {
+      properties: true,
+      deactivation: true,
+      memberPlan: true,
+      periods: {
+        include: {
+          Invoice: true
+        }
+      }
+    }
+  })
+}
+
+async function deletePeriodOfUnpaidInvoice(
+  subscriptionPeriodClient: PrismaClient['subscriptionPeriod'],
+  subscription: Subscription,
+  invoice: Invoice
+) {
+  return subscriptionPeriodClient.deleteMany({
+    where: {
+      invoiceID: invoice.id
+    }
+  })
+}
+
+async function deleteUnpaidInvoices(
+  invoiceClient: PrismaClient['invoice'],
+  subscriptionPeriodClient: PrismaClient['subscriptionPeriod'],
+  subscription: Subscription
+) {
+  const unpaidInvoices = await invoiceClient.findMany({
+    where: {
+      subscriptionID: subscription.id,
+      paidAt: null
+    }
+  })
+  for (const unpaidInvoice of unpaidInvoices) {
+    await deletePeriodOfUnpaidInvoice(subscriptionPeriodClient, subscription, unpaidInvoice)
+    await invoiceClient.delete({
+      where: {
+        id: unpaidInvoice.id
+      }
+    })
+  }
+}
+
+export class PayrexxSubscriptionPaymentProvider extends BasePaymentProvider {
+  readonly instanceName: string
+  readonly instanceAPISecret: string
+  readonly webhookSecret: string
+
+  constructor(props: PayrexxSubscripionsPaymentProviderProps) {
+    super(props)
+    this.instanceName = props.instanceName
+    this.instanceAPISecret = props.instanceAPISecret
+    this.webhookSecret = props.webhookSecret
+    this.activateHook(props.prisma)
+  }
+
+  activateHook(prisma: PrismaClient) {
+    console.log('Activate Payrexx subscription prisma hook')
+
+    const hook = (): Prisma.Middleware => async (params, next) => {
+      // Only handle subscription update db queries skip all other
+      if (params.model !== 'Subscription' || params.action !== 'update') {
+        return next(params)
+      }
+
+      const subscription = params.args.data
+      const properties: MetadataProperty[] = params.args.data.properties.createMany.data
+      const deactivation = params.args.data.deactivation
+
+      // Get paymentProviderName
+      const paymentProvider = await prisma.paymentMethod.findUnique({
+        where: {
+          id: subscription.paymentMethodID
+        }
+      })
+
+      // Exit if subscription has no external id or is other payment provider
+      if (!subscription || paymentProvider?.paymentProviderID !== this.id) {
+        return next(params)
+      }
+
+      // Find external id property and fail if subscription has been deactivated
+      const isPayrexxExt = properties.find(sub => sub.key === 'payrexx_external_id')
+      if (!isPayrexxExt) {
+        throw Error("It's not supported to reactivate a deactivated Payrexx subscription!")
```

Comment:
We should handle that also in the editor UI.

Reply:
We can make this in a later change (Issue) if necessary but its important to get this through in the next release.